src/jit/morph.cpp

```diff
@@ -7411,6 +7411,41 @@ void Compiler::fgMorphTailCallViaHelper(GenTreeCall* call, void* pfnCopyArgs)
     if (call->IsVirtualStub())
     {
         call->gtFlags |= GTF_CALL_NULLCHECK;
+
+#if defined(_TARGET_AMD64_)
+        // If we already inited arg info here then we will have added the VSD
+        // arg on AMD64. So we remove it here as we will handle this case
```

Comment:
Hmm, I'm not familiar with the rest of the code. Does the special handling account for the fact that an arg was removed but the corresponding `fgArgTabEntry` still points to it?

Reply:
Yes, this function changes the arg list in other ways that invalidates that info. It is cleared after the call here:
https://github.com/dotnet/coreclr/blob/3983c45a5f4dca49a4543d8d28635005868c18d9/src/jit/morph.cpp#L7250-L7257